### ISSUE TITLE
Cura OSX crashing on startup

### ISSUE THREAD

**Issue by AletheianAlex** [Ultimaker/Cura#1668]
Cura 2.4.0 crashes on startup about 2 out of 3 times, with the same crash report and log results every time. I have tried re-installing Cura (including manually removing all pref files and configurations) as well as trying several Mac OS X versions (10.12.3 & 10.12.4) and several different user accounts.   I may have to be downgrade for now to v2.3.1, which hardly ever crashed.  

My details: 2010 Mac Pro with a fresh install of OS X 10.12.4 and a GTX680 cpu with Nvidia web drivers and Qt version Version: 7.7.3.  Here are the crash reports and Cura.log file:

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CURA.LOG:

2017-04-09 14:48:57,800 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin FileLogger
2017-04-09 14:48:57,804 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin GCodeProfileReader
2017-04-09 14:48:57,808 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin GCodeWriter
2017-04-09 14:48:57,812 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin ImageReader
2017-04-09 14:48:57,818 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin LayerView
2017-04-09 14:48:57,822 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin LegacyProfileReader
2017-04-09 14:48:57,827 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin LocalFileOutputDevice
2017-04-09 14:48:57,832 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin MachineSettingsAction
2017-04-09 14:48:57,836 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin PerObjectSettingsTool
2017-04-09 14:48:57,841 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin PostProcessingPlugin
2017-04-09 14:48:57,857 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin RemovableDriveOutputDevice
2017-04-09 14:48:57,868 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin SliceInfoPlugin
2017-04-09 14:48:57,873 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin SolidView
2017-04-09 14:48:57,877 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin CameraTool
2017-04-09 14:48:57,884 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin MirrorTool
2017-04-09 14:48:57,894 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin RotateTool
2017-04-09 14:48:57,901 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin ScaleTool
2017-04-09 14:48:57,906 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin SelectionTool
2017-04-09 14:48:57,911 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin TranslateTool
2017-04-09 14:48:57,918 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin UltimakerMachineActions
2017-04-09 14:48:57,934 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin UM3NetworkPrinting
2017-04-09 14:48:57,945 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin UpdateChecker
2017-04-09 14:48:57,945 - INFO - UpdateChecker.UpdateChecker.run [40]: Checking for new version of cura
2017-04-09 14:48:57,985 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin USBPrinting
2017-04-09 14:48:57,992 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin VersionUpgrade21to22
2017-04-09 14:48:57,996 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin VersionUpgrade22to24
2017-04-09 14:48:58,001 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin SimpleView
2017-04-09 14:48:58,006 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin WireframeView
2017-04-09 14:48:58,011 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin X3DReader
2017-04-09 14:48:58,023 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin XmlMaterialProfile
2017-04-09 14:48:58,028 - INFO - UM.PluginRegistry.loadPlugin [100]: Loaded plugin XRayView
2017-04-09 14:48:58,030 - INFO - UM.Qt.QtApplication.__init__ [90]: Command line arguments: {'external-backend': False, 'disable-textures': False, 'file': []}
2017-04-09 14:48:58,049 - INFO - UM.VersionUpgradeManager.upgrade [100]: Looking for old configuration files to upgrade.
2017-04-09 14:48:58,055 - WARNING - UM.Logger.logException [65]: Exception: Exception in machine_instance upgrade with VersionUpgrade22to24.VersionUpgrade: Traceback (most recent call last):
  File "/Users/ultimaker/build/2.4/build/inst/lib/python3.5/site-packages/UM/VersionUpgradeManager.py", line 310, in _upgradeFile
TypeError: 'NoneType' object is not iterable

2017-04-09 14:48:58,057 - WARNING - UM.Logger.logException [69]: Traceback (most recent call last):
2017-04-09 14:48:58,058 - WARNING - UM.Logger.logException [69]:   File "/Users/ultimaker/build/2.4/build/inst/lib/python3.5/site-packages/UM/VersionUpgradeManager.py", line 310, in _upgradeFile
2017-04-09 14:48:58,060 - WARNING - UM.Logger.logException [69]: TypeError: 'NoneType' object is not iterable
2017-04-09 14:48:58,063 - WARNING - UM.Logger.logException [65]: Exception: Exception in machine_instance upgrade with VersionUpgrade22to24.VersionUpgrade: Traceback (most recent call last):
  File "/Users/ultimaker/build/2.4/build/inst/lib/python3.5/site-packages/UM/VersionUpgradeManager.py", line 310, in _upgradeFile
TypeError: 'NoneType' object is not iterable

2017-04-09 14:48:58,065 - WARNING - UM.Logger.logException [69]: Traceback (most recent call last):
2017-04-09 14:48:58,067 - WARNING - UM.Logger.logException [69]:   File "/Users/ultimaker/build/2.4/build/inst/lib/python3.5/site-packages/UM/VersionUpgradeManager.py", line 310, in _upgradeFile
2017-04-09 14:48:58,068 - WARNING - UM.Logger.logException [69]: TypeError: 'NoneType' object is not iterable
2017-04-09 14:48:58,077 - DEBUG - UM.Backend.Backend._logSocketState [167]: Socket state changed to Listening
2017-04-09 14:48:58,087 - INFO - UM.Backend.Backend.startEngine [71]: Started engine process: /Applications/Cura.app/Contents/MacOS/CuraEngine
2017-04-09 14:48:58,088 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] Calling engine with: ['/Applications/Cura.app/Contents/MacOS/CuraEngine', 'connect', '127.0.0.1:49674', '-j', '/Applications/Cura.app/Contents/MacOS/resources/definitions/fdmprinter.def.json', '']
2017-04-09 14:48:58,091 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] 
2017-04-09 14:48:58,092 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] Cura_SteamEngine version DEV
2017-04-09 14:48:58,094 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] Copyright (C) 2014 David Braam
2017-04-09 14:48:58,096 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] 
2017-04-09 14:48:58,097 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] This program is free software: you can redistribute it and/or modify
2017-04-09 14:48:58,100 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] it under the terms of the GNU Affero General Public License as published by
2017-04-09 14:48:58,104 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] the Free Software Foundation, either version 3 of the License, or
2017-04-09 14:48:58,106 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] (at your option) any later version.
2017-04-09 14:48:58,107 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] 
2017-04-09 14:48:58,109 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] This program is distributed in the hope that it will be useful,
2017-04-09 14:48:58,110 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] but WITHOUT ANY WARRANTY; without even the implied warranty of
2017-04-09 14:48:58,112 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
2017-04-09 14:48:58,115 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] GNU Affero General Public License for more details.
2017-04-09 14:48:58,117 - DEBUG - UM.Backend.Backend._logSocketState [171]: Socket state changed to Connected
2017-04-09 14:48:58,118 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] 
2017-04-09 14:48:58,120 - DEBUG - UM.Backend.Backend._onSocketStateChanged [161]: Backend connected on port 49674
2017-04-09 14:48:58,121 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] You should have received a copy of the GNU Affero General Public License
2017-04-09 14:48:58,128 - DEBUG - UM.Backend.Backend._backendLog [89]: [Backend] along with this program.  If not, see <http://www.gnu.org/licenses/>.
2017-04-09 14:48:58,595 - DEBUG - MachineSettingsAction.MachineSettingsAction._onContainerAdded [69]: Not attaching MachineSettingsAction to ord; Multi-extrusion printers are not supported
2017-04-09 14:48:58,806 - DEBUG - MachineSettingsAction.MachineSettingsAction._onContainerAdded [69]: Not attaching MachineSettingsAction to ultimaker_original_dual; Multi-extrusion printers are not supported
2017-04-09 14:48:59,869 - DEBUG - MachineSettingsAction.MachineSettingsAction._onContainerAdded [69]: Not attaching MachineSettingsAction to ultimaker3; Multi-extrusion printers are not supported
2017-04-09 14:49:00,297 - DEBUG - MachineSettingsAction.MachineSettingsAction._onContainerAdded [69]: Not attaching MachineSettingsAction to ultimaker3_extended; Multi-extrusion printers are not supported
2017-04-09 14:49:00,792 - DEBUG - MachineSettingsAction.MachineSettingsAction._onContainerAdded [69]: Not attaching MachineSettingsAction to punchtec_connect_xl; Multi-extrusion printers are not supported
2017-04-09 14:49:06,658 - WARNING - UM.Settings.InstanceContainer.addMetaDataEntry [172]: Meta data with key compatible was already added.
2017-04-09 14:49:09,845 - DEBUG - UM.Controller.setActiveView [81]: Setting active view to SolidView
2017-04-09 14:49:09,851 - WARNING - cura.Settings.ExtruderManager.getMachineExtruders [375]: Tried to get the extruder trains for machine MpSelectMini, which doesn't exist.
2017-04-09 14:49:09,853 - WARNING - cura.Settings.ExtruderManager.getMachineExtruders [375]: Tried to get the extruder trains for machine MpSelectMini, which doesn't exist.
2017-04-09 14:49:10,392 - DEBUG - UM.Qt.Bindings.Theme.load [126]: Loading theme file: /Applications/Cura.app/Contents/MacOS/resources/themes/cura/theme.json
2017-04-09 14:49:10,484 - DEBUG - UM.Qt.Bindings.Theme.load [186]: Loaded theme /Applications/Cura.app/Contents/MacOS/resources/themes/cura
2017-04-09 14:49:11,171 - DEBUG - UM3NetworkPrinting.DiscoverUM3Action._createAdditionalComponentsView [134]: Creating additional ui components for UM3.
2017-04-09 14:49:11,239 - DEBUG - CuraEngineBackend.CuraEngineBackend.slice [152]: Starting slice job...


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CRASH REPORT:

Process:               cura [826]
Path:                  /Applications/Cura.app/Contents/MacOS/cura
Identifier:            nl.ultimaker.cura
Version:               2.4 (2.4.0)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           cura [826]
User ID:               501

Date/Time:             2017-04-09 14:49:11.461 -0400
OS Version:            Mac OS X 10.12.4 (16E195)

Time Awake Since Boot: 9800 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x0000000618a4b180
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Bus error: 10
Termination Reason:    Namespace SIGNAL, Code 0xa
Terminating Process:   exc handler [0]

VM Regions Near 0x618a4b180:
    MALLOC_LARGE           0000000618a0c000-0000000618a4a000 [  248K] rw-/rwx SM=PRV  
--> JS JIT generated code  0000000618a4a000-0000000618a4c000 [    8K] r-x/rwx SM=PRV  
    ATS (font support)     0000000618b24000-000000061aaa4000 [ 31.5M] rw-/rwx SM=COW  

Thread 0 Crashed:
0   ???                           	0x0000000618a4b180 0 + 26183250304
1   libQt5Qml.5.dylib             	0x0000000604fd4fe7 QV4::Script::run() + 279
2   libQt5Qml.5.dylib             	0x0000000604f32795 QJSEngine::evaluate(QString const&, QString const&, int) + 229
3   PyQt5.QtQml.so                	0x00000006047dc1b6 meth_QJSEngine_evaluate(_object*, _object*, _object*) + 278
4   nl.ultimaker.cura             	0x00000001024a7e4c PyCFunction_Call + 60
5   nl.ultimaker.cura             	0x00000001025252e0 PyEval_EvalFrameEx + 25584
6   nl.ultimaker.cura             	0x000000010252a2c2 0x102452000 + 885442
7   nl.ultimaker.cura             	0x0000000102524c2f PyEval_EvalFrameEx + 23871
8   nl.ultimaker.cura             	0x0000000102529ad2 0x102452000 + 883410
9   nl.ultimaker.cura             	0x000000010251eece PyEval_EvalCodeEx + 78
10  nl.ultimaker.cura             	0x000000010248844d 0x102452000 + 222285
11  nl.ultimaker.cura             	0x000000010245ed95 PyObject_Call + 101
12  nl.ultimaker.cura             	0x0000000102474add 0x102452000 + 142045
13  nl.ultimaker.cura             	0x000000010245ed95 PyObject_Call + 101
14  nl.ultimaker.cura             	0x00000001024bdade 0x102452000 + 441054
15  nl.ultimaker.cura             	0x00000001024ba099 0x102452000 + 426137
16  nl.ultimaker.cura             	0x000000010245ed95 PyObject_Call + 101
17  nl.ultimaker.cura             	0x0000000102528a45 PyEval_CallObjectWithKeywords + 165
18  sip.so                        	0x0000000604604a96 sip_api_call_method + 211
19  PyQt5.QtQml.so                	0x000000060480e5d6 QPyQmlObjectProxy::createPyObject(QObject*) + 118
20  PyQt5.QtQml.so                	0x00000006047fbd40 void QQmlPrivate::createInto<QPyQmlObject11>(void*) + 16
21  libQt5Qml.5.dylib             	0x000000060505a505 QQmlType::create() const + 69
22  libQt5Qml.5.dylib             	0x00000006050bf6d5 QQmlObjectCreator::createInstance(int, QObject*, bool) + 949
23  libQt5Qml.5.dylib             	0x00000006050c31f8 QQmlObjectCreator::setPropertyBinding(QQmlPropertyData const*, QV4::CompiledData::Binding const*) + 904
24  libQt5Qml.5.dylib             	0x00000006050c0cee QQmlObjectCreator::setupBindings(QBitArray const&) + 1230
25  libQt5Qml.5.dylib             	0x00000006050c4e36 QQmlObjectCreator::populateInstance(int, QObject*, QObject*, QQmlPropertyData const*, QBitArray const&) + 1270
26  libQt5Qml.5.dylib             	0x00000006050c00e8 QQmlObjectCreator::createInstance(int, QObject*, bool) + 3528
27  libQt5Qml.5.dylib             	0x00000006050bf24b QQmlObjectCreator::create(int, QObject*, QQmlInstantiationInterrupt*) + 699
28  libQt5Qml.5.dylib             	0x00000006050bfa4a QQmlObjectCreator::createInstance(int, QObject*, bool) + 1834
29  libQt5Qml.5.dylib             	0x00000006050bf24b QQmlObjectCreator::create(int, QObject*, QQmlInstantiationInterrupt*) + 699
30  libQt5Qml.5.dylib             	0x000000060504e7e0 QQmlIncubatorPrivate::incubate(QQmlInstantiationInterrupt&) + 208
31  libQt5Qml.5.dylib             	0x000000060504e5cd QQmlEnginePrivate::incubate(QQmlIncubator&, QQmlContextData*) + 445
32  libQt5Qml.5.dylib             	0x000000060504bc65 QQmlComponent::create(QQmlIncubator&, QQmlContext*, QQmlContext*) + 485
33  libQt5Quick.5.dylib           	0x00000006067f4a01 QQuickLoaderPrivate::_q_sourceLoaded() + 689
34  libQt5Qml.5.dylib             	0x00000006050c552f QQmlObjectCreator::finalize(QQmlInstantiationInterrupt&) + 687
35  libQt5Qml.5.dylib             	0x000000060504ec6f QQmlIncubatorPrivate::incubate(QQmlInstantiationInterrupt&) + 1375
36  libQt5Qml.5.dylib             	0x000000060504e5cd QQmlEnginePrivate::incubate(QQmlIncubator&, QQmlContextData*) + 445
37  libQt5Qml.5.dylib             	0x00000006050e86d8 QQmlDelegateModelPrivate::object(QQmlListCompositor::Group, int, bool) + 1528
38  libQt5Quick.5.dylib           	0x0000000606821d9b QQuickItemViewPrivate::createItem(int, bool) + 203
39  libQt5Quick.5.dylib           	0x00000006067cd341 QQuickListViewPrivate::addVisibleItems(double, double, double, double, bool) + 769
40  libQt5Quick.5.dylib           	0x0000000606822418 QQuickItemViewPrivate::refill(double, double) + 264
41  libQt5Quick.5.dylib           	0x000000060681ee6e QQuickItemViewPrivate::layout() + 958
42  libQt5Quick.5.dylib           	0x000000060674196e QQuickWindowPrivate::polishItems() + 238
43  libQt5Quick.5.dylib           	0x000000060670f869 QSGGuiThreadRenderLoop::renderWindow(QQuickWindow*) + 681
44  PyQt5.QtQuick.so              	0x000000060655d9ce sipQQuickWindow::exposeEvent(QExposeEvent*) + 126
45  libQt5Gui.5.dylib             	0x00000006048f18e5 QWindow::event(QEvent*) + 501
46  libQt5Quick.5.dylib           	0x0000000606748999 QQuickWindow::event(QEvent*) + 633
47  PyQt5.QtQuick.so              	0x000000060655d674 sipQQuickWindow::event(QEvent*) + 116
48  libQt5Widgets.5.dylib         	0x0000000605b28a7a QApplicationPrivate::notify_helper(QObject*, QEvent*) + 314
49  libQt5Widgets.5.dylib         	0x0000000605b29eae QApplication::notify(QObject*, QEvent*) + 510
50  PyQt5.QtWidgets.so            	0x0000000605587b6f sipQApplication::notify(QObject*, QEvent*) + 127
51  libQt5Core.5.dylib            	0x000000060416eaf4 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 164
52  libQt5Gui.5.dylib             	0x00000006048e6654 QGuiApplicationPrivate::processExposeEvent(QWindowSystemInterfacePrivate::ExposeEvent*) + 308
53  libQt5Gui.5.dylib             	0x00000006048cb574 QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 452
54  libQt5Gui.5.dylib             	0x00000006048c855c QWindowSystemInterface::flushWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 572
55  libqcocoa.dylib               	0x000000060850a6e3 QCocoaWindow::setVisible(bool) + 707
56  libQt5Gui.5.dylib             	0x00000006048ee065 QWindow::setVisible(bool) + 1045
57  libQt5Gui.5.dylib             	0x00000006048f3697 QWindow::qt_metacall(QMetaObject::Call, int, void**) + 71
58  libQt5Quick.5.dylib           	0x0000000606752878 QQuickWindow::qt_metacall(QMetaObject::Call, int, void**) + 24
59  PyQt5.QtQuick.so              	0x000000060655c948 sipQQuickWindow::qt_metacall(QMetaObject::Call, int, void**) + 24
60  PyQt5.QtQml.so                	0x000000060480e4f2 QPyQmlObjectProxy::qt_metacall(QMetaObject::Call, int, void**) + 242
61  libQt5Qml.5.dylib             	0x0000000605046318 QQmlPropertyPrivate::write(QObject*, QQmlPropertyData const&, QVariant const&, QQmlContextData*, QFlags<QQmlPropertyPrivate::WriteFlag>) + 968
62  libQt5Qml.5.dylib             	0x0000000605010611 QV4::QObjectWrapper::setProperty(QV4::ExecutionEngine*, QObject*, QQmlPropertyData*, QV4::Value const&) + 3857
63  libQt5Qml.5.dylib             	0x00000006050299db QV4::Runtime::setQmlQObjectProperty(QV4::ExecutionEngine*, QV4::Value const&, int, QV4::Value const&) + 171
64  ???                           	0x0000000616497875 0 + 26143717493
65  libQt5Qml.5.dylib             	0x0000000604fb4bde QV4::SimpleScriptFunction::call(QV4::Managed const*, QV4::CallData*) + 606
66  libQt5Qml.5.dylib             	0x00000006050af2ac QQmlJavaScriptExpression::evaluate(QV4::CallData*, bool*) + 716
67  libQt5Qml.5.dylib             	0x0000000605055d5e QQmlBoundSignalExpression::evaluate(void**) + 958
68  libQt5Qml.5.dylib             	0x000000060505649d QQmlBoundSignal_callback(QQmlNotifierEndpoint*, void**) + 445
69  libQt5Qml.5.dylib             	0x000000060508f385 QQmlNotifier::emitNotify(QQmlNotifierEndpoint*, void**) + 741
70  libQt5Core.5.dylib            	0x00000006041a12b9 QMetaObject::activate(QObject*, int, int, void**) + 121
71  libQt5Qml.5.dylib             	0x000000060510cbf5 QQmlTimer::event(QEvent*) + 149
72  libQt5Widgets.5.dylib         	0x0000000605b28a7a QApplicationPrivate::notify_helper(QObject*, QEvent*) + 314
73  libQt5Widgets.5.dylib         	0x0000000605b29eae QApplication::notify(QObject*, QEvent*) + 510
74  PyQt5.QtWidgets.so            	0x0000000605587b6f sipQApplication::notify(QObject*, QEvent*) + 127
75  libQt5Core.5.dylib            	0x000000060416eaf4 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 164
76  libQt5Core.5.dylib            	0x000000060416f85b QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 891
77  libqcocoa.dylib               	0x000000060851ea4e QCocoaEventDispatcherPrivate::processPostedEvents() + 190
78  libqcocoa.dylib               	0x000000060851f301 QCocoaEventDispatcherPrivate::postedEventsSourceCallback(void*) + 33
79  com.apple.CoreFoundation      	0x00007fff796413b1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
80  com.apple.CoreFoundation      	0x00007fff7962263c __CFRunLoopDoSources0 + 556
81  com.apple.CoreFoundation      	0x00007fff79621b26 __CFRunLoopRun + 934
82  com.apple.CoreFoundation      	0x00007fff79621524 CFRunLoopRunSpecific + 420
83  com.apple.HIToolbox           	0x00007fff78b81ebc RunCurrentEventLoopInMode + 240
84  com.apple.HIToolbox           	0x00007fff78b81cf1 ReceiveNextEventCommon + 432
85  com.apple.HIToolbox           	0x00007fff78b81b26 _BlockUntilNextEventMatchingListInModeWithFilter + 71
86  com.apple.AppKit              	0x00007fff7711ce24 _DPSNextEvent + 1120
87  com.apple.AppKit              	0x00007fff7789885e -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 2796
88  com.apple.AppKit              	0x00007fff771117ab -[NSApplication run] + 926
89  libqcocoa.dylib               	0x000000060851e21f QCocoaEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 2191
90  libQt5Core.5.dylib            	0x000000060416ace1 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 417
91  libQt5Core.5.dylib            	0x000000060416f165 QCoreApplication::exec() + 341
92  PyQt5.QtWidgets.so            	0x0000000605588730 meth_QApplication_exec_(_object*, _object*) + 80
93  nl.ultimaker.cura             	0x00000001024a7eb5 PyCFunction_Call + 165
94  nl.ultimaker.cura             	0x00000001025252e0 PyEval_EvalFrameEx + 25584
95  nl.ultimaker.cura             	0x000000010252a2c2 0x102452000 + 885442
96  nl.ultimaker.cura             	0x0000000102524c2f PyEval_EvalFrameEx + 23871
97  nl.ultimaker.cura             	0x0000000102529ad2 0x102452000 + 883410
98  nl.ultimaker.cura             	0x000000010251ee71 PyEval_EvalCode + 81
99  nl.ultimaker.cura             	0x000000010251c6c0 0x102452000 + 829120
100 nl.ultimaker.cura             	0x00000001024a7eb5 PyCFunction_Call + 165
101 nl.ultimaker.cura             	0x00000001025252e0 PyEval_EvalFrameEx + 25584
102 nl.ultimaker.cura             	0x0000000102529ad2 0x102452000 + 883410
103 nl.ultimaker.cura             	0x000000010251ee71 PyEval_EvalCode + 81
104 nl.ultimaker.cura             	0x0000000102453ae4 main + 1428
105 nl.ultimaker.cura             	0x0000000102453544 start + 52

**Comment by nallath** [comment]
It seems like you have a corrupted local folder. Could you try to delete those files and try again?

**Comment by AletheianAlex** [comment]
I gave it a go, but it made no difference.  I did a total uninstall/preference-wipe/profile-delete and reinstall again just to be sure.  Finally to test, I did another complete uninstall, trashed prefs and deleted profiles, etc, and reinstalled 2.3.1 and it works fine.  I then trashed everything again and installed the 2.5 beta, and that version worked fine.  Trashed everything yet again and reinstalled 2.4 and it crashes constantly like before.  So for now I will be on the 2.5 beta, and if that acts up in a future patch I'll downgrade to 2.3.1.
